### PR TITLE
Empty actionx

### PR DESCRIPTION
**Pull request by joakim-hove**
Allow ACTIONX keywords with empty conditions, these ACTIONX keywords will always
evaluate to false. Main motivation is to facilitate dummy keywords for PYACTION.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/input/eclipse/Schedule/Action/ActionAST.cpp [hunk 1]
```diff
@@ -47,9 +47,11 @@ AST AST::serializeObject()
...
 
 Action::Result AST::eval(const Action::Context& context) const {
-    if (this->condition)
+    if (this->condition) {
+        if (this->condition->empty())
+            return Action::Result(false);
         return this->condition->eval(context);
-    else
+    } else
         return Action::Result(false);
```
Comment (bska) [inline src/opm/input/eclipse/Schedule/Action/ActionAST.cpp:55]: Personally I'd probably combine these conditions to reduce nesting
```C++
if (this->condition && !this->condition->empty())
    return this->condition->eval(context);
else
    return Action::Result(false);
```
or possibly reversed for that matter
```C++
if (!this->condition || this->condition->empty())
    return Action::Result(false);
else
    return this->condition->eval(context);
```

**Review by bska — APPROVED** [review]
This looks good to me.  If you'd like to consider my suggestion that's fine too, but otherwise please just merge this.

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/input/eclipse/Schedule/Action/ActionAST.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/input/eclipse/Schedule/Action/ActionAST.cpp]: I have updated to use the second alternative.

File: src/opm/input/eclipse/Schedule/Action/ActionAST.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ AST AST::serializeObject()
 }
 
 Action::Result AST::eval(const Action::Context& context) const {
-    if (this->condition)
+    if (!this->condition || this->condition->empty())
         return this->condition->eval(context);
     else
         return Action::Result(false);
```
Comment (bska) [inline src/opm/input/eclipse/Schedule/Action/ActionAST.cpp:53]: Ehm, you probably meant to either switch the `return` statements or reverse the branch condition here.

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/input/eclipse/Schedule/Action/ActionAST.cpp (hunk 2 above)
Comment (joakim-hove) [reply src/opm/input/eclipse/Schedule/Action/ActionAST.cpp]: Ehmmm - yes!

**Review by bska — APPROVED** [review]
Thanks for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**